app/src/main/java/io/apicurio/registry/rest/v2/GroupsResourceImpl.java

```diff
@@ -469,14 +480,22 @@ public void deleteArtifactsInGroup(String groupId) {
     }
 
     /**
-     * @see io.apicurio.registry.rest.v2.GroupsResource#createArtifact(java.lang.String, io.apicurio.registry.types.ArtifactType, java.lang.String, java.lang.String, io.apicurio.registry.rest.v2.beans.IfExists, java.lang.Boolean, java.io.InputStream)
+     * @see io.apicurio.registry.rest.v2.GroupsResource#createArtifact(String, ArtifactType, String, String, IfExists, Boolean, String, String, String, String, InputStream)
      */
     @Override
-    @Authorized(style=AuthorizedStyle.GroupOnly, level=AuthorizedLevel.Write)
-    public ArtifactMetaData createArtifact(String groupId,
-            ArtifactType xRegistryArtifactType, String xRegistryArtifactId, String xRegistryVersion,
-            IfExists ifExists, Boolean canonical, InputStream data) {
+    @Authorized(style = AuthorizedStyle.GroupOnly, level = AuthorizedLevel.Write)
+    public ArtifactMetaData createArtifact(String groupId, ArtifactType xRegistryArtifactType, String xRegistryArtifactId,
+                                           String xRegistryVersion, IfExists ifExists, Boolean canonical,
+                                           String xRegistryDescription, String xRegistryDescriptionEncoded,
+                                           String xRegistryName, String xRegistryNameEncoded, InputStream data) {
         requireParameter("groupId", groupId);
+
+        maxOneOf("xRegistryName", xRegistryName, "xRegistryNameEncoded", xRegistryNameEncoded);
+        maxOneOf("xRegistryDescription", xRegistryDescription, "xRegistryDescriptionEncoded", xRegistryDescriptionEncoded);
```

Comment:
I think the names you pass in here should be the actual HTTP header names:
* X-Registry-Name
* X-Registry-Name-Encoded
* X-Registry-Description
* X-Registry-Description-Encoded

Same for all of the calls to `maxOneOf` elsewhere.